tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,171 @@
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+# regex that matches server/client 'write to' with the TLS header
+OPENSSL_WRITE_TO_REGEX = r"write to [\w >\(\)\[\]=]*\\n0000 - 17 03 03 [0-9A-Fa-f]{2} [0-9A-Fa-f]{2}"
+
+
+def get_payload_size_from_openssl_trace(record: str) -> int:
+    # last five characters in trace is the record size in the form XX XX where X is a hex digit
+    size_in_hex = record[-5:].replace(' ', '')
+    size = int(size_in_hex, 16)
+    # record includes 16 bytes of aead tag
+    return size - 16
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_server_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                           padding_size):
+    # Communication Timeline
+    # Client [OpenSSL]                       | Server [S2N]
+    # Handshake                              | Handshake
+    # Send padded data                       | Receive padded data
+    # Send padded close connection           | Close
+    port = next(available_ports)
+
+    random_bytes = data_bytes(PAYLOAD_SIZE)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        cert=certificate.cert,
+        curve=curve,
+        data_to_send=random_bytes,
+        insecure=True,
+        protocol=protocol,
+        extra_flags=['-record_padding', padding_size]
+    )
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.extra_flags = None
+
+    s2nd = managed_process(S2N, server_options, timeout=5)
+    openssl = managed_process(provider, client_options, timeout=5)
+
+    for client_results in openssl.get_results():
+        client_results.assert_success()
+
+        # find all occurrences of openssl writing to server
+        all_occurrences = re.findall(
+            OPENSSL_WRITE_TO_REGEX, str(client_results.stdout))
+
+        for occurrence in all_occurrences:
+            payload_size = get_payload_size_from_openssl_trace(occurrence)
+            # all sent records must be padded to padding_size
+            assert payload_size > 0
+            assert payload_size % padding_size == 0
```

Comment:
This should be the same for both tests, so should probably be in a common method.